src/ImageSharp/Common/Helpers/Shuffle/IComponentShuffle.cs

```diff
@@ -97,15 +99,15 @@ public void RunFallbackShuffle(ReadOnlySpan<byte> source, Span<byte> dest)
         {
             ref uint sBase = ref Unsafe.As<byte, uint>(ref MemoryMarshal.GetReference(source));
             ref uint dBase = ref Unsafe.As<byte, uint>(ref MemoryMarshal.GetReference(dest));
-            int n = source.Length / 4;
+            nuint n = (nuint)(uint)source.Length / 4;
 
-            for (int i = 0; i < n; i++)
+            for (nuint i = 0; i < n; i++)
             {
-                uint packed = Unsafe.Add(ref sBase, i);
+                uint packed = Unsafe.Add(ref sBase, (int)i);
```

Comment:
```suggestion
                uint packed = Unsafe.Add(ref sBase, i);
```
The whole point of `nint` here is to avoid a sign extending move (`movsxd`)* in generated machine code. `nint` has the native size of the interge, i.e. it's like a `int` on 32-bit, and a `long` on 64-bit.
The cast back to `(int)` is actually a de-optimization. Just keep the `nint`.

\* this is a rather slow one, especially compared to normal `mov`-operations.
See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdwAJ4YYDALIAKWADMG12w1wAhPDDQMvDyBAJQMALwAfAwAqly42C6CAIIAJqnOMG4+fgG8IRrU5hbuNnZOrqWeObj+DFxBGKER0XEJSSxpGZU1dfmFxTpV5ZluHna9AVwcjc1RsfGJKemjXr61eQU0AL5AA==).

Reply:
Ok, wasn't aware of this difference between the different machine code calls. Usin `nint`'s now